athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchMetadataHandler.java

```diff
@@ -187,6 +195,14 @@ public ListTablesResponse doListTables(BlockAllocator allocator, ListTablesReque
 
                     indices.add(new TableName(request.getSchemaName(), index));
                 }
+
```

Comment:
github won't let me select this whole function, but you can rewrite and simplify this entire method:

```
    @Override
    public ListTablesResponse doListTables(BlockAllocator allocator, ListTablesRequest request)
            throws RuntimeException
    {
        logger.debug("doListTables: enter - " + request);
        String endpoint = getDomainEndpoint(request.getSchemaName());
        AwsRestHighLevelClient client = clientFactory.getOrCreateClient(endpoint);
        try {
            Stream<String> indices = client.getAliases().stream().filter(index -> !index.startsWith("."));
            // handle data stream, 1 data stream can contains multiple indices which start with ".ds-xxxxxxxxxx"
            Stream<String> dataStreams = client.indices()
                .getDataStream(new GetDataStreamRequest("*"), RequestOptions.DEFAULT)
                .getDataStreams()
                .stream()
                .map(dataStream -> dataStream.getName());
            List<TableName> tableNames = Stream.concat(indices, dataStreams)
                .map(tableName -> new TableName(request.getSchemaName(), tableName))
                .collect(Collectors.toList());
            return new ListTablesResponse(request.getCatalogName(), tableNames, null);
        }
        catch (IOException error) {
            throw new RuntimeException("Error retrieving indices: " + error.getMessage(), error);
        }
    }
```